src/http/common.cr

```diff
@@ -117,17 +160,57 @@ module HTTP
     right_index = bytesize
     if middle_index >= right_index
       return {name, ""}
-    elsif right_index > 1 && cstr[right_index - 2] === '\r' && cstr[right_index - 1] === '\n'
+    elsif right_index > 1 && cstr[right_index - 2] == '\r'.ord.to_u8 && cstr[right_index - 1] == '\n'.ord.to_u8
       right_index -= 2
-    elsif right_index > 0 && cstr[right_index - 1] === '\n'
+    elsif right_index > 0 && cstr[right_index - 1] == '\n'.ord.to_u8
       right_index -= 1
     end
 
-    value = line.byte_slice(middle_index, right_index - middle_index)
+    value = String.new(slice[middle_index, right_index - middle_index])
 
     {name, value}
   end
 
+  # :nodoc:
+  COMMON_HEADERS = %w(
```

Comment:
No. A tuple creates a new type for the compiler for no reason. Tuples main use case is returning multiple values from a method. All other uses are a smell.

Reply:
Thanks for this tip.
But here, having an Array allocates to the heap, right? Or there is an optimization I am not aware of.
Or perhaps a `StaticArray`?

Reply:
It allocates once for the entire duration of the program. It's negligible. It's a problem if you allocate an array in a tight loop, or even more than once when you could use a tuple. But if it's just once then Array is the best option.

Reply:
Thanks for the explanation.
However, having an Array constant is mutable (we can append elements to it), but not a Tuple constant.
This one can be made private.

Reply:
Done private, thanks.

The mutability argument is good. But I still think in this case it doesn't matter because it's used in just one line of code.